src/main/java/io/lettuce/core/event/jfr/EventRecorder.java

```diff
@@ -38,16 +39,6 @@ static EventRecorder getInstance() {
      */
     RecordableEvent start(Event event);
 
-    /**
-     * Interface defining a recordable event that is recorded on calling {@link #record()}.
-     */
-    interface RecordableEvent {
-
-        /**
-         * Complete the event recording.
-         */
-        void record();
-
-    }
+    void publish(Event event);
```

Comment:
What is the semantic difference between `record(Event)` and `publish(Event)`? 

Such an arrangement asks for trouble because the methods are ambiguous without clear distinction.

`RecordableEvent` is part of the API, moving it around introduces a breaking change without good reason. All necessary changes are possible without breaking the public API.

Reply:
I can move `RecordableEvent` back to `RecordableEvent`.


I added publish per https://github.com/redis/lettuce/pull/2819#issuecomment-2044326923, specifically:
> * Adopt `JfrEventRecorder.publish(…)` to check whether the given event is `EventRecorder.RecordableEvent`. If so, then we call `record(…)` on the event instead of `jdk.jfr.Event jfrEvent = createEvent(…);`

Agreed that they create ambiguity. Should I combine the implementation of `JfrEventRecorder.#record` and `JfrEventRecorder#publish` into `#record`?